### PR TITLE
Refactoring the way GC Stress logic is used while running tests

### LINKED ISSUE #5245: [RTM] R2R GCStress jobs on Unix aren't actually running under GCStress
This line: https://github.com/dotnet/coreclr/blob/master/tests/src/CLRTest.Execute.Bash.targets#L22 is using
```
export complus_gcstress=0xF
```
Which has no effect on Unix because environment variables are case sensitive.  Proper capitalization is 
```
export COMPlus_GCStress=0xF
```

### PR DESCRIPTION
**Pull request by ramarag**
A separate test build is no longer needed for GC stress test Runs

Fixes #5245

@gkhanna79 @RussKeldorph  @swgillespie @wtgodbe PTAL

### CODE REVIEW

**Comment by ramarag** [comment]
@dotnet-bot test ci

**Comment by ramarag** [comment]
@dotnet-bot test ci please

**Comment by wtgodbe** [comment]
LGTM

**Comment by ramarag** [comment]
@dotnet-bot test ci please

**Comment by ramarag** [comment]
@dotnet-bot  Windows_NT x86 ryujit Checked Build and Test

**Comment by ramarag** [comment]
@dotnet-bot  test Windows_NT x86 ryujit Checked Build and Test

File: tests/runtest.cmd [hunk 1]
```diff
@@ -350,10 +351,11 @@ echo TestEnv- Optional parameter - this will run a custom script to set custom t
 echo VSVersion- Optional parameter - VS2013 or VS2015 ^(default: VS2015^)
 echo GenerateLayoutOnly - If specified will not run the tests and will only create the Runtime Dependency Layout
 echo RunCrossgenTests   - Runs ReadytoRun tests
-echo jitstress <n>      - Runs the tests with COMPlus_JitStress=n
-echo jitstressregs <n>  - Runs the tests with COMPlus_JitStressRegs=n
```
Comment (RussKeldorph) [inline tests/runtest.cmd]: I think these can be escape with caret (e.g. ^<) if you prefer to keep them.

**Comment by swgillespie** [comment]
LGTM

**Comment by RussKeldorph** [comment]
The title of this change and the description seem to be very wrong.  Otherwise LGTM.

**Pull request merged**